### ISSUE TITLE
Any way to get blob of image

### ISSUE THREAD

**Issue by youniaogu** [georstat/react-native-image-cache#39]
Hi, i have a question, is any way to get blob of image, i need to handle source image with canvas after fetch 

Is possible implement on `CacheManager.prefetch` or some way? For example:

```
CacheManager.prefetch(url).then((blob) => {})
```

Thank you.

**Comment by efstathiosntonas** [comment]
Hi @youniaogu sure, we'll add it.

**Comment by efstathiosntonas** [comment]
@youniaogu are you using typescript or javascript in your project? 

If you're using typescript, can you add this at line ~166 on `CacheManager.ts` and let me know if it's what you need:

```
  static async prefetchBlob(
    source: string | string[],
    options?: DownloadOptions
  ): Promise<string> {
    if (typeof source === "string") {
      const path = await CacheManager.get(source, options).getPath();
      const blob = await FileSystem.readFile(path, "base64");
      return blob;
    }
  }
```

Call it with ```
CacheManager.prefetchBlob(YOUR IMAGE URL).then((response) =>
      console.log(response)
    );
```

**Comment by youniaogu** [comment]
> @youniaogu are you using typescript or javascript in your project?
> 
> If you're using typescript, can you add this at line ~166 on `CacheManager.ts` and let me know if it's what you need:
> 
> ```ts
>   static async prefetchBlob(
>     source: string,
>     options?: DownloadOptions
>   ): Promise<string> {
>     if (typeof source === "string") {
>       const path = await CacheManager.get(source, options).getPath();
>       const blob = await FileSystem.readFile(path, "base64");
>       return blob;
>     }
>   }
> ```
> 
> Call it with
> 
> ```ts
> CacheManager.prefetchBlob(YOUR IMAGE URL).then((response) =>
>       console.log(response)
>     );
> ```

thank you for quick reply, i am using typescript

it look work, i am trying

**Comment by efstathiosntonas** [comment]
@youniaogu 

with correct typings: 

```
 static async prefetchBlob(
    source: string,
    options?: DownloadOptions
  ): Promise<string | undefined> {
    const path = await CacheManager.get(source, options).getPath();
    if (path) {
      const blob = await FileSystem.readFile(path, 'base64');
      return blob;
    }
    return undefined;
  }
```

If it works, please use `patch-package` and patch it until we release a version that supports this.

**Comment by youniaogu** [comment]
@efstathiosntonas It work, thank you

**Comment by efstathiosntonas** [comment]
ok great, I'll release a new version in a while including this.

**Comment by efstathiosntonas** [comment]
@youniaogu released [2.3.0 ](https://github.com/georstat/react-native-image-cache/releases/tag/v2.3.0)
